src/openrct2/world/Footpath.cpp

```diff
@@ -696,48 +691,51 @@ static TileElement* footpath_get_element(int32_t x, int32_t y, int32_t z0, int32
 /**
  * Attempt to connect a newly disconnected queue tile to the specified path tile
  */
-static bool footpath_reconnect_queue_to_path(int32_t x, int32_t y, TileElement* tileElement, int32_t action, int32_t direction)
+static bool footpath_reconnect_queue_to_path(
+    const CoordsXY& footpathPos, TileElement* tileElement, int32_t action, int32_t direction)
 {
     if (((tileElement->AsPath()->GetEdges() & (1 << direction)) == 0) ^ (action < 0))
         return false;
 
-    int32_t x1 = x + CoordsDirectionDelta[direction].x;
-    int32_t y1 = y + CoordsDirectionDelta[direction].y;
+    auto shiftedFootpathPos = footpathPos + CoordsDirectionDelta[direction];
 
     if (action < 0)
     {
-        if (fence_in_the_way(x, y, tileElement->base_height, tileElement->clearance_height, direction))
+        if (fence_in_the_way({ footpathPos, tileElement->GetBaseZ(), tileElement->GetClearanceZ() }, direction))
             return false;
 
-        if (fence_in_the_way(x1, y1, tileElement->base_height, tileElement->clearance_height, direction_reverse(direction)))
+        if (fence_in_the_way(
+                { shiftedFootpathPos, tileElement->GetBaseZ(), tileElement->GetClearanceZ() }, direction_reverse(direction)))
             return false;
     }
 
-    int32_t z = tileElement->base_height;
-    TileElement* otherTileElement = footpath_get_element(x1, y1, z - 2, z, direction);
+    int32_t z = tileElement->GetBaseZ();
+    TileElement* otherTileElement = footpath_get_element({ shiftedFootpathPos, z - LAND_HEIGHT_STEP, z }, direction);
```

Comment:
`targetFootpathElement`. I would then make a new variable called `targetQueueElement` after confirming it is not a nullptr and that it is a queue. Then we can get rid of all of the `AsPath()`